tests/python/contrib/test_arm_compute_lib/test_add.py

```diff
@@ -0,0 +1,142 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Arm Compute Library integration reshape tests."""
+
+import numpy as np
+
+import tvm
+import tvm.testing
+from tvm import relay
+
+from test_arm_compute_lib.infrastructure import (
+    skip_runtime_test,
+    skip_codegen_test,
+    build_and_run,
+    verify,
+    verify_codegen,
+)
+from test_arm_compute_lib.infrastructure import Device
+
+
+def _get_model(shape, dtype, var_names, op, op_params):
+    a = relay.var(next(var_names), shape=shape, dtype=dtype)
+    b = relay.var(next(var_names), shape=shape, dtype=dtype)
+    return op(a, b, **op_params)
+
+
+def _get_expected_codegen(shape, dtype):
+    node = {
+        "op": "kernel",
+        "name": "add",
+        "inputs": [[0, 0, 0], [1, 0, 0]],
+        "attrs": {
+            "num_inputs": "2",
+            "num_outputs": "1",
+            "shape": [[list(shape)]],
+            "dtype": [[dtype]],
+        },
+    }
+
+    input_a = {"op": "input", "name": "", "attrs": {"shape": [[list(shape)]], "dtype": [[dtype]]}}
+    input_b = {"op": "input", "name": "", "attrs": {"shape": [[list(shape)]], "dtype": [[dtype]]}}
+    return [input_a, input_b, node]
+
+
+def test_add():
+    Device.load("test_config.json")
+
+    if skip_runtime_test():
+        return
+
+    device = Device()
+    np.random.seed(0)
+
+    for dtype, low, high, atol, rtol, op, op_params in [
+        ("float32", -127, 128, 1e-7, 1e-7, relay.add, {}),
+        # different qnn params
+        (
+            "uint8",
+            0,
+            255,
+            0.0,
+            1.0,
+            relay.qnn.op.add,
+            {
+                "lhs_scale": relay.const(0.0156863, "float32"),
+                "lhs_zero_point": relay.const(127, "int32"),
+                "rhs_scale": relay.const(0.0117647, "float32"),
+                "rhs_zero_point": relay.const(85, "int32"),
+                "output_scale": relay.const(0.0235294, "float32"),
+                "output_zero_point": relay.const(128, "int32"),
+            },
+        ),
+        # same qnn params
+        (
+            "uint8",
+            0,
+            255,
+            0.0,
+            1.0,
+            relay.qnn.op.add,
+            {
+                "lhs_scale": relay.const(0.0126863, "float32"),
+                "lhs_zero_point": relay.const(127, "int32"),
+                "rhs_scale": relay.const(0.0126863, "float32"),
+                "rhs_zero_point": relay.const(127, "int32"),
+                "output_scale": relay.const(0.0126863, "float32"),
+                "output_zero_point": relay.const(127, "int32"),
+            },
+        ),
+    ]:
```

Comment:
1. This is hard to follow. If possible, could you make a list of 3 elements and iterate the list here?
2. Could you elaborate the purpose of testing 2 QNN adds?

Reply:
1. Done.
2. Removed